src/utils/hooks.tsx

```diff
@@ -18,3 +27,27 @@ export function useTestData(): [string | null, TestData | 'NO_DATA' | null] {
   const test_data = get_test(test_id);
   return [test_id, test_data];
 }
+
+export function safeJsonParse(string: string | null, defaultValue?: any): JSONValue {
+  if (!string) return defaultValue;
+  try {
+    return JSON.parse(string);
+  } catch {
+    return defaultValue;
+  }
+}
+
+export function useLocalStorageState<T = any>(
+  key: string,
+  initialValue?: JSONValue<T>
+): [typeof state, typeof setState] {
+  const [state, setState] = useState<T>(
+    () => safeJsonParse(localStorage.getItem(key)) || initialValue
```

Comment:
You define `defaultValue` as an argument of the function `safeJsonParse`, but ignore it here and instead use `|| initialValue`, which will handle edge cases such as falsy JSON values (such as `null` or `[]`) differently. 

I am not saying this is the wrong way, it's just something that should probably be cleaned up before merging.